codec-http2/src/main/java/io/netty/handler/codec/http2/HpackEncoder.java

```diff
@@ -380,126 +403,157 @@ private HeaderEntry getEntryInsensitive(CharSequence name, CharSequence value) {
      * Returns the lowest index value for the header field name in the dynamic table. Returns -1 if
      * the header field name is not in the dynamic table.
      */
-    private int getIndex(CharSequence name) {
-        if (length() == 0 || name == null) {
-            return NOT_FOUND;
+    private NameEntry getEntry(CharSequence name) {
+        if (isEmpty() || name == null) {
+            return null;
         }
         int h = AsciiString.hashCode(name);
-        int i = index(h);
-        for (HeaderEntry e = headerFields[i]; e != null; e = e.next) {
+        for (NameEntry e = nameEntries[bucket(h)]; e != null; e = e.next) {
             if (e.hash == h && equalsConstantTime(name, e.name) != 0) {
```

Comment:
@carl-mastrangelo any reason to use constant time header name comparison? Isn't it beneficial from security perspective only for header values?

Reply:
The RFC is kinda vague on this.  It (rfc7541#section-6.2.3) mentions the never-indexed bit is for fields (name+value).  Later in the same definition it says:

> This representation is intended for protecting header field values that are not to be put at risk by compressing them.

However, it also seems to mean that the name literal (for prefix 0 0 0 1 ) should not be indexed, implying it may be sensitive?   I think it's reasonable to use the insensitive compare, but I can't quote line and verse to back it up.